### ISSUE TITLE
Backprop causes NaN

### ISSUE THREAD

**Issue by ndahlquist** [wuhuikai/DeepGuidedFilter#7]
Using your Pytorch implementation, inference works great (output images look beautiful). Unfortunately, if I try to use this layer as part of a model and run an optimizer through it, I'm getting NaN gradients. I can try to debug this, but curious if you have seen this issue before?

**Comment by ndahlquist** [comment]
Okay the problem was that I was setting `eps` to 0 is the constructor; my mistake!
